core/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/WorkspacePermissionsFilter.java

```diff
@@ -79,7 +79,6 @@ public void filter(GenericResourceMethod genericResourceMethod, Object[] argumen
                 workspaceId = ((String)arguments[0]);
                 action = DELETE;
                 break;
-            case "recoverWorkspace":
```

Comment:
What about https://github.com/codenvy/codenvy/pull/448/files/5a4a3a9ff8769b39a48d21dabedac33cb45aa167#diff-15c913e41deb296c99e1a0ffcb8a5937L141 ?
Do you mean this mechanism?